core/server/helpers/body_class.js

```diff
@@ -14,44 +14,45 @@ var hbs             = require('express-hbs'),
     template        = require('./template'),
     body_class;
 
-body_class = function () {
+body_class = function (options) {
     var classes = [],
+        context = options.data.root.context,
         post = this.post,
         tags = this.post && this.post.tags ? this.post.tags : this.tags || [],
         page = this.post && this.post.page ? this.post.page : this.page || false;
 
-    if (this.tag !== undefined) {
-        classes.push('tag-template');
-        classes.push('tag-' + this.tag.slug);
-    }
-
-    if (this.author !== undefined) {
-        classes.push('author-template');
-        classes.push('author-' + this.author.slug);
-    }
-
-    if (_.isString(this.relativeUrl) && this.relativeUrl.match(/\/(page\/\d)/)) {
-        classes.push('paged');
-        // To be removed from pages by #2597 when we're ready to deprecate this
-        classes.push('archive-template');
-    } else if (!this.relativeUrl || this.relativeUrl === '/' || this.relativeUrl === '') {
-        classes.push('home-template');
-    } else if (post) {
+    if (post) {
         // To be removed from pages by #2597 when we're ready to deprecate this
-        // i.e. this should be if (post && !page) { ... }
+        // i.e. this should be if (_.contains(context, 'post') && post) { ... }
         classes.push('post-template');
     }
 
-    if (page) {
+    if (_.contains(context, 'home')) {
```

Comment:
Switched `body_class` to be context-driven rather than having the same (roughly) rules duplicated, as has been done for `meta_title`.